### ISSUE TITLE
Display an error message when queryRenderedFeatures is passed a non-existantant layer id

### ISSUE THREAD

**Issue by BrunoSalerno** [mapbox/mapbox-gl-js#3084]
When passing to `queryRenderedFeatures` layers that don't exist, I get the following error:
```
Cannot read property 'source' of undefined
    at Object.queryRenderedFeatures (https://api.tiles.mapbox.com/mapbox-gl-js/v0.23.0/mapbox-gl.js:104:10526)
    at queryRenderedFeatures (https://api.tiles.mapbox.com/mapbox-gl-js/v0.23.0/mapbox-gl.js:192:5443)
```
It looks like it is not being checked that the layer does exist. 

**mapbox-gl-js version**: 0.23.0

### Steps to Trigger Behavior

`var features = map.queryRenderedFeatures(point, {layers: ['not_existent_layer']})`

### Expected Behavior

I expect this code not to trigger an error. Instead, I should get an empty array.

### Actual Behavior

I get an error.

**Comment by peterqliu** [comment]
An empty array communicates that:

- the requested layer exists, and
- there are no features within the queried area/point

Expanding it to include nonexistent layers adds ambiguity, makes it harder to detect situations when you accidentally mistype the layer name. 

An error is the appropriate response here, though it could be more helpful about the point of failure (`"Requested layer does not exist"`).

**Comment by mollymerp** [comment]
fixed in #3196
